pkg/utils/utils.go

```diff
@@ -137,42 +139,36 @@ func RetrieveVcConfigSecret(params map[string]interface{}, config *rest.Config,
 		return errors.New(errMsg)
 	}
 
-	var sEnc string
-	var lines []string
+	ParseConfig(secret, params, logger)
+
+	return nil
+}
+
+func ParseConfig(secret *k8sv1.Secret, params map[string]interface{}, logger logrus.FieldLogger) {
+	// Setup config
+	var conf vcConfig.Config
+	// Read secret data
 	for _, value := range secret.Data {
-		sEnc = string(value)
-		lines = strings.Split(sEnc, "\n")
-		logger.Debugf("Successfully retrieved vCenter configuration from secret %s", secret.Name)
+		confStr := string(value) // Convert the secret data to a string
+		err := gcfg.FatalOnly(gcfg.ReadStringInto(&conf, confStr))
+		if err != nil {
+			logger.WithError(err).Error("Failed to parse vSphere secret data")
+		}
```

Comment:
It seems like you have decided to ignore this error. I don't think it's correct.

Reply:
If it's preferred, I can refactor ParseConfig to return an error and set the call of ParseConfig as:
```
err := ParseConfig(secret, params, logger)
if err != nil {
    return err
}

return nil
```

Reply:
I think the error would make a lot of sense, early fail rather than waiting till the actual vc connect.